### PR TITLE
[SPARK-12256] [SQL] Code refactoring: naming boolean variables

### PR DESCRIPTION
**Pull request by gatorsmile**
Hard to read the source codes sometimes if the boolean parameter values are unnamed.

Not sure if this is desirable to the others. If this does not make sense, feel free to close it. 

Thank you! @marmbrus @rxin @cloud-fan

### CODE REVIEW

**Comment by gatorsmile** [comment]
retest this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala [hunk 1]
```diff
@@ -172,7 +172,8 @@ object JavaTypeInference {
       .getOrElse(UnresolvedAttribute(part))
 
     /** Returns the current path or `BoundReference`. */
-    def getPath: Expression = path.getOrElse(BoundReference(0, inferDataType(typeToken)._1, true))
+    def getPath: Expression = path.getOrElse(BoundReference(0, inferDataType(typeToken)._1,
+      nullable = true))
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala]: i wouldn't wrap this here ... i'd just put the entire body of the function on its own line

**Comment by rxin** [comment]
@gatorsmile  can you close this pull request? Knowing when to wrap is pretty important. I think most wrappings you have done are suboptimal here.

**Comment by marmbrus** [comment]
+1 to @rxin concerns on wrapping.  A good rule of thumb is to always break at the highest syntatic level (not in the middle of some construct like a list of arguments).  Otherwise you break things up that are actually the same and create an artificial separation.

```scala
// No
def getPath: Expression = path.getOrElse(BoundReference(0, inferDataType(typeToken)._1,
  nullable = true))

// Yes
def getPath: Expression = 
  path.getOrElse(BoundReference(0, inferDataType(typeToken)._1, nullable = true))
```

**Comment by gatorsmile** [comment]
@rxin Sure, let me close it. 
@marmbrus Thank you for explaining the desired Scala wrapping style! Sorry for spending time on it. 

In the product code I am working on is completely different, each parameter is always in different lines. For example, 
```scala
def getPath: Expression = 
  path.getOrElse(BoundReference(0, 
                                inferDataType(typeToken)._1, 
                                nullable = true))
```

**Comment by marmbrus** [comment]
We allow each parameter to be on its own line (though only if they are long enough to warrant that).  However we prefer to keep the code closer to the left margin.
```scala
def getPath: Expression = 
  path.getOrElse(
    BoundReference(
      0, 
      inferDataType(typeToken)._1, 
      nullable = true))
```

**Comment by gatorsmile** [comment]
Uh, I see. Thanks again!

**Pull request closed without merging**